ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -2165,6 +2165,17 @@ def create_active_user(host, login, password, first='test', last='user',
     kdestroy_all(host)
 
 
+def set_user_password(host, username, password):
+    temppass = "redhat\nredhat"
+    sendpass = f"redhat\n{password}\n{password}"
+    kdestroy_all(host)
+    kinit_admin(host)
+    host.run_command(["ipa", "passwd", username],stdin_text=temppass)
+    host.run_command(["kinit", username], stdin_text=sendpass)
+    kdestroy_all(host)
+    kinit_admin(host)
```

Comment:
It would be more generic to take the current or 'old' password as an argument. This assumes it will be 'redhat'.

I'd drop the final kinit_admin(). The caller may not expect it, unless the code pattern is that we'd always do a kinit_admin() anyway.

Reply:
Are you talking about generalizing the function to be used by non-admin users too?

I used the 'redhat' password because it's forcing the change as admin so it only prompts for the new password.  Right?  Or, are there situations I'm not thinking of where changing the password as admin could prompt for current password?

As for the final kinit admin, that has been my standard pattern for tests but, I can see where it could cause issues.  Safer to leave it to the person using it to have to kinit as needed afterwards.